Formula/sh4d0wup.rb

```diff
@@ -0,0 +1,28 @@
+class Sh4d0wup < Formula
+  desc "Signing-key abuse and update exploitation framework"
+  homepage "https://github.com/kpcyrd/sh4d0wup"
+  url "https://github.com/kpcyrd/sh4d0wup/archive/v0.9.0.tar.gz"
+  sha256 "ec6aa007417e76338bc903d969e88f038ba982eb603f227f85caea5b8ad715c3"
+  license "GPL-3.0-or-later"
+
+  depends_on "llvm@15" => :build # for libclang
+  depends_on "pkg-config" => :build
+  depends_on "rust" => :build
+  depends_on "nettle"
+  depends_on "openssl@3"
+  depends_on "pcsc-lite"
+  depends_on "zstd"
+
+  def install
+    system "cargo", "install", *std_cargo_args
+
+    generate_completions_from_executable(bin/"sh4d0wup", "completions")
+  end
+
+  test do
+    system bin/"sh4d0wup", "keygen", "tls"
+    system bin/"sh4d0wup", "keygen", "pgp"
+    system bin/"sh4d0wup", "keygen", "ssh", "--type=ed25519", "--bits=256"
+    system bin/"sh4d0wup", "keygen", "openssl", "--secp256k1"
+  end
```

Comment:
can we also assert the response in here?

Also we probably need linkage test as well

```
  test do
    linked_libraries = [
      Formula["openssl@3"].opt_lib/shared_library("libssl"),
      Formula["openssl@3"].opt_lib/shared_library("libcrypto"),
    ]
    linked_libraries << (Formula["openssl@3"].opt_lib/shared_library("libcrypto")) if OS.mac?

    linked_libraries.each do |library|
      assert check_binary_linkage(bin/"sh4d0wup", library),
             "No linkage with #{library.basename}! Cargo is likely using a vendored version."
    end
  end
```

Reply:
> Is there a way to pipe the output of my other commands into a program (like pgpdump or openssl x509)? I'm not sure how to do this in ruby. :)

I think I got it to work! :3